modules/services/preferences/src/main/java/au/com/shiftyjelly/pocketcasts/preferences/Settings.kt

```diff
@@ -174,17 +178,51 @@ interface Settings {
             fun itemForId(id: String): MediaNotificationControls? {
                 return items[id]
             }
-        }
-
-        object Archive : MediaNotificationControls(LR.string.archive, IR.drawable.ic_archive, ARCHIVE_KEY)
-
-        object MarkAsPlayed : MediaNotificationControls(LR.string.mark_as_played, IR.drawable.ic_markasplayed, MARK_AS_PLAYED_KEY)
 
-        object PlayNext : MediaNotificationControls(LR.string.play_next, com.google.android.gms.cast.framework.R.drawable.cast_ic_mini_controller_skip_next, PLAY_NEXT_KEY)
+            fun fromServerId(id: String) = All.find { it.serverId == id }
```

Comment:
We should replace `find` with `first` to reduce the unnecessary iterations.

Reply:
`find()` and `first()` are not equivalent. `first()` throws an exception. If you meant `firstOrNull()` [like here](https://github.com/Automattic/pocket-casts-android/pull/1759#discussion_r1471050482) it is the same as `find()`. It is just a semantic difference.